service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/auth/TokenCredentialCacheTest.java

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package tests.unit.com.microsoft.azure.sdk.iot.service.auth;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.microsoft.azure.sdk.iot.service.auth.TokenCredentialCache;
+import com.sun.scenario.effect.Offset;
+import mockit.Expectations;
+import mockit.Mocked;
+import org.joda.time.field.OffsetDateTimeField;
+import org.junit.Test;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+import static org.junit.Assert.assertEquals;
+
+public class TokenCredentialCacheTest
+{
+    @Mocked
+    TokenCredential mockTokenCredential;
+
+    @Mocked
+    AccessToken mockAccessToken;
+
+    @Mocked
+    AccessToken mockAccessToken2;
+
+    @Mocked
+    Mono<AccessToken> mockTokenTask;
+
+    @Test
+    public void tokenCredentialCachesToken()
+    {
+        TokenCredentialCache cache = new TokenCredentialCache(mockTokenCredential);
+
+        new Expectations()
+        {
+            {
+                mockTokenCredential.getToken((TokenRequestContext) any).block();
+                result = mockAccessToken;
+
+                mockAccessToken.getExpiresAt();
+                result = OffsetDateTime.MAX;
+            }
+        };
+
+        AccessToken accessToken = cache.getAccessToken();
+        AccessToken accessToken2 = cache.getAccessToken();
+
+        assertEquals(mockAccessToken, accessToken);
+        assertEquals(mockAccessToken, accessToken2);
+    }
+
+    @Test
+    public void tokenCredentialProactivelyRenewsToken()
+    {
+        TokenCredentialCache cache = new TokenCredentialCache(mockTokenCredential);
```

Comment:
Do you need line 63 to 74? It seems like it is the same as what was tested in the previous method.

Reply:
The expectations look the same, but then this test changes the expectations before attempting to get the next token. I do need these lines to mimic the typical first generation of an access token. The next expectations make it so that the renewal logic kicks in based off of the expiry time of the mock access token